File: e2e/playwright-typescript/interactions/pages/TaskInfoModal.ts
import { Locator, Page } from '@playwright/test';
import { TaskPriority } from '@/data/TaskPriority';
import { TaskStatus } from '@/data/TaskStatus';

export class TaskInfoModal {
  readonly title: Locator;
  readonly description: Locator;
  readonly validationBadge: Locator;

  constructor(private readonly page: Page) {
    this.title = page.getByTestId('modal-title');
    this.description = page.getByTestId('description');
    this.validationBadge = page.getByTestId('valid');
  }

  statusTag(status: TaskStatus): Locator {
    return this.page.getByTestId('status').getByTestId(`status-tag-${status}`);
  }

  priorityTag(priority: TaskPriority): Locator {
    return this.page.getByTestId('priority').getByTestId(`priority-tag-${priority}`);
  }
}

File: e2e/playwright-python/interactions/pages/task_info_modal.py
from playwright.sync_api import Page, Locator
from data.task_status import TaskStatus
from data.task_priority import TaskPriority


class TaskInfoModal:
    def __init__(self, page: Page):
        self._page = page
        self.title: Locator = page.get_by_test_id("modal-title")
        self.description: Locator = page.get_by_test_id("description")
        self.validation_badge: Locator = page.get_by_test_id("valid")

    def status_tag(self, status: TaskStatus) -> Locator:
        return self._page.get_by_test_id("status").get_by_test_id(f"status-tag-{status.value}")

    def priority_tag(self, priority: TaskPriority) -> Locator:
        return self._page.get_by_test_id("priority").get_by_test_id(f"priority-tag-{priority.value}")
